client/src/components/SubmissionTabs/SubmissionCard/SubmissionCard.tsx

```diff
@@ -9,50 +9,72 @@ import Button from '@material-ui/core/Button';
 import Box from '@material-ui/core/Box';
 import DoneIcon from '@material-ui/icons/Done';
 
-import { useAuth } from '../../../context/useAuthContext';
-import { Contest } from '../../../interface/Contest';
 import useStyles from './useStyles';
 import SubmissionCardProps from '../SumissionCardInterface';
-import { fetchContestById } from '../../../helpers/APICalls/searchContest';
+import { Contest } from '../../../interface/Contest';
+import { User } from '../../../interface/User';
+import { Winner } from '../../../interface/Winner';
+import { postWinner } from '../../../helpers/APICalls/winners';
+import chargeCustomer from '../../../helpers/APICalls/chargeCustomer';
+import { useSnackBar } from '../../../context/useSnackbarContext';
+import deadlinePassed from '../../../helpers/datePassed';
 
 export interface ParamProps {
   id: string;
 }
 
-function SubmissionCard({ imageSrc, author }: SubmissionCardProps): JSX.Element {
+function SubmissionCard({ imageSrc, author, contest }: SubmissionCardProps): JSX.Element {
   const classes = useStyles();
   const { loggedInUser } = useAuth();
-  const [contest, setContest] = useState<Contest>();
-  const { id } = useParams<ParamProps>();
-
-  useEffect(() => {
-    const ac = new AbortController();
-    fetchContestById({ id: id }).then((res) => {
-      if (res.success) {
-        setContest(res.success as Contest);
-      }
-    });
-    return ac.abort();
-  }, [id]);
+  const { updateSnackBarMessage } = useSnackBar();
 
   if (loggedInUser === undefined || !loggedInUser || contest == undefined || contest.userID == undefined)
     return <CircularProgress />;
 
+  const handleWinnerSelection = async (
+    imageSrc: string,
+    author: User,
+    contest: Contest,
+    user: User | null | undefined,
```

Comment:
Why are we using null/undefined types here?

Reply:
user types are coming from the loggedinUser. If I just change it to user. It's throwing type error.